### ISSUE TITLE
RELICENSE pyzmq core

### ISSUE THREAD

**Issue by minrk** [zeromq/pyzmq#1039]
Following libzmq (https://github.com/zeromq/libzmq/issues/2376), I'd like to attempt to relicense away from LGPL. That issue is largely copied here, with small substitutions for BSD instead of MPLv2, but MPLv2 is left as an option.

Unlike libzmq itself, the preferred new license for pyzmq is [3-Clause BSD](https://opensource.org/licenses/BSD-3-Clause) because the majority of pyzmq is already licensed under BSD. Only the low-level Cython bindings are licensed under LGPL as a historical artifact, so only those who have contributed to that part of pyzmq (zmq.backend.cython, zmq.core, zmq._zmq, depending on the point in history) are required to submit grants in order for this process to complete.

We are requesting contributors to send in a license grant to relicense pyzmq contributions to one of the options listed below:

1. relicense to the MPLv2 or any Open Source Initiative (OSI) approved license chosen by the current pyzmq BDFL **(preferred selection)**
1. relicense to the BSD license specifically
1. relicense to the MPLv2 license specifically (only choose this if you object to permissive licenses)

If a contribution was made during working hours for an employer then we may also need a license grant (or "quit claim") from the employer.

We would like to follow a certain template as license grant to ensure all the necessary information is included.  There are three templates, one for each option.  Please choose the template that best represents your wishes.

The templates can be found at: https://github.com/zeromq/pyzmq/tree/HEAD/RELICENSE/templates

To include the license grant, please:
- fork the original pyzmq repository
- create a new file in the RELICENSE directory, named after your name and/or your employers name, with the license grant as content, using the template chosen
- commit this file to the repository, using the same email address as your earlier contributions
- create a pull request on pyzmq to have your license grant merged into the repository.

See the existing files in:
https://github.com/zeromq/pyzmq/tree/HEAD/RELICENSE

or this specific example: https://github.com/zeromq/pyzmq/blob/HEAD/RELICENSE/minrk.md

for a guide for how to fill out the template.

---

If there are any questions, don't hesitate to ask them below.

**Comment by Jeducious** [comment]
Hi Min, I am in the process of packaging a new version of our software which uses pyzmq. I have read the relicense readme and this issue. But, I am still unclear as to what I should include in my package? You state that you want to switch to 3-claus-BSD, but then the template that is preferred is either MPLv2 or 'other'. 

I hope you can appreciate that to the not so experienced licensee, this is rather confusing. I obviously want to comply, but all the responses from the various authors don't specify what license they are providing. There has also been no decree from the BDFL as to what OSI license you have adopted. 

Can you please update this so that myself and others wishing to build on py zmq can have clarity over the licensing obligations we have?

Thanks!

**Comment by minrk** [comment]
@Jeducious since the relicense process is not complete, currently parts of pyzmq core are still LGPL and the rest is BSD. Based on responses, it will likely be MPLv2, though I would have preferred BSD-3.

> Can you please update this so that myself and others wishing to build on py zmq can have clarity over the licensing obligations we have?

The pyzmq licenses apply no restrictions to folks "building on pyzmq," depending on what you mean. From my interpretation (and intention) of the license, all use of pyzmq unmodified is acceptable, without restriction. It is only when you distribute *modified versions of pyzmq* that the LGPL copyleft restrictions of "derived works" come into effect. To be clear: `import zmq` Python code *does not* constitute creating a derived work for the purposes of the LGPL here.

**Comment by Jeducious** [comment]
Hi Min, thanks! That has cleared it up for us.

**Comment by capfei** [comment]
Is the relicensing still in progress? Also, I noticed that the LICENSE.LESSER contains GPL 3.0 text instead of LGPL 3.0. I'm assuming it should be LGPL since I see that mentioned through out the repo.

**Comment by minrk** [comment]
still in progress, yes. After reaching out to developers, I got no response from some, and can't really complete it, and don't have the energy to chase people down. There've been ~no LGPL contributions in about 5 years, so I'm tempted to try to strip and replace the few remaining lines and finish it.

Every relicense approval allowed BSD, so that's what I'll go with if I ever have the energy to finish.

Thanks for noticing the text! A supposed file rename accidentally used the wrong text. Fixed now.
